src/shared/dissect-image.c

```diff
@@ -4050,13 +4051,18 @@ int verity_dissect_and_mount(
 
         relax_extension_release_check = mount_options_relax_extension_release_checks(options);
 
-        /* We might get an FD for the image, but we use the original path to look for the dm-verity files */
-        r = verity_settings_load(&verity, src, NULL, NULL);
-        if (r < 0)
-                return log_debug_errno(r, "Failed to load root hash: %m");
+        /* We might get an FD for the image, but we use the original path to look for the dm-verity files.
+         * The caller might also give us a pre-loaded VeritySettings, in which case we just use it. */
```

Comment:
maybe add a comment here that stresses that we'll also extend VeritySettings with more info because dissected_image_load_verity_sig_partition() is invoked.